addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+    l10n_tw_edi_carrier_type = fields.Selection(
+        selection=[("1", "ECpay e-invoice carrier"), ("2", "Citizen Digital Certificate"), ("3", "Mobile Barcode")],
+        string="Carrier Type",
+        copy=False,
+        compute="_compute_carrier_info",
+        store=True,
+        readonly=False)
+    l10n_tw_edi_carrier_number = fields.Char(string="Carrier Number", copy=False, compute="_compute_carrier_info", store=True, readonly=False)
+    l10n_tw_edi_invoice_amount = fields.Float(string="Invoice Amount", readonly=True, copy=False)
+    l10n_tw_edi_invoice_shop_custom_number = fields.Char(string="Cooperative Shop Custom Number", readonly=True, copy=False)
+    l10n_tw_edi_invoice_create_date = fields.Datetime(string="Creation Date", readonly=True, copy=False)
+    l10n_tw_edi_refund_finish = fields.Boolean(string="Refund Finish", readonly=True, copy=False)
+    l10n_tw_edi_refund_state = fields.Selection(
+        selection=[
+            ("draft", "draft"),
+            ("to be agreed", "To be agreed"),
+            ("agreed", "Agreed"),
+            ("disagree", "Disagree"),
+        ],
+        string="Refund State",
+        default="draft",
+        readonly=True,
+        copy=False,
+    )
+    l10n_tw_edi_refund_agreement_type = fields.Selection(
+        [("offline", "Offline Agreement"), ("online", "Online Agreement")],
+        default="offline",
+        string="Refund invoice Agreement Type",
+        required=True,
+        copy=False,
+    )
+    l10n_tw_edi_origin_invoice_number = fields.Many2one('account.move', string="Original Invoice", readonly=True, copy=False)
+    l10n_tw_edi_refund_invoice_number = fields.Char(string="Refund Invoice Number", readonly=True, copy=False)
+    l10n_tw_edi_remain_refundable_amount = fields.Float(string="Remain refundable amount", readonly=True, copy=False)
+
+    @api.depends('l10n_tw_edi_is_donate', 'l10n_tw_edi_carrier_type')
+    def _compute_is_print(self):
+        for move in self:
+            if move.l10n_tw_edi_is_donate or move.l10n_tw_edi_carrier_type:
+                move.l10n_tw_edi_is_print = False
+
+    @api.depends('l10n_tw_edi_is_print', 'l10n_tw_edi_carrier_type')
+    def _compute_is_donate(self):
+        for move in self:
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_carrier_type:
+                move.l10n_tw_edi_is_donate = False
+                move.l10n_tw_edi_love_code = False
+
+    @api.depends('l10n_tw_edi_is_print', 'l10n_tw_edi_is_donate')
+    def _compute_carrier_info(self):
+        for move in self:
+            if move.l10n_tw_edi_is_print or move.l10n_tw_edi_is_donate:
+                move.l10n_tw_edi_carrier_type = False
+                move.l10n_tw_edi_carrier_number = False
+
+    @api.depends('line_ids.tax_ids')
+    def _compute_is_included_tax(self):
+        for move in self:
+            product_lines = move.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines())
```

Comment:
You cannot call _get_downpayment_lines in this module, it comes from sales. I'm not sure it's even needed

Same thing below